lib/board.rb

```diff
@@ -0,0 +1,62 @@
+require "pry"
+class Board
+  attr_accessor :cells, :board
+  
+  def reset!
+    @cells = Array.new(9, " ") 
+  end
+  
+  def initialize(board = nil)
+    @cells = Array.new(9," ")
+  end
+  
+  def display
+  puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} \n-----------\n #{@cells[3]} | #{@cells[4]} | #{@cells[5]} \n-----------\n #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+  
+  def position(input)
+    value = (input.to_i)-1
+    cell = @cells[value]
+    cell
+  end
+  
+  def full?
+    @cells.all? do |box|
+      box == "X" || box == "O"
+    end
+  end
+  
+  def turn_count
+    count = 0 
+    @cells.each do |i|
+      if i == "X" || i == "O"
+       count += 1 
+      end
+    end
+   return count  
+  end
+  
+  def taken?(index)
+    if position(index) == "X" || position(index) == "O"
+      return true
+    end
+    if position(index) == " " || position(index) == ""
+      return false
+    end
+  end
```

Comment:
We can use Ruby's implicit return to simplify this down to just one line:
```
def taken?(index)
  position(index) == "X" || position(index) == "O"
end
```

This also applies to other methods, including `#valid_move?`, `#current_player`, `#draw`, and `#over`.